vividus-extension-selenium/src/main/java/org/vividus/proxy/har/HarOnFailureManager.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2019-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.proxy.har;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+import com.browserup.harreader.model.Har;
+import com.browserup.harreader.model.HarEntry;
+import com.browserup.harreader.model.HarLog;
+
+import org.vividus.proxy.IProxy;
+import org.vividus.testcontext.TestContext;
+
+public class HarOnFailureManager
+{
+    private final TestContext testContext;
+    private IProxy proxy;
+
+    public HarOnFailureManager(IProxy proxy, TestContext testContext)
+    {
+        this.testContext = testContext;
+        this.proxy = proxy;
+    }
+
+    public Optional<Har> takeHar()
+    {
+        return Optional.of(createHarWithUniqueData(proxy.getRecordedData()));
+    }
+
+    private Har createHarWithUniqueData(Har har)
+    {
+        Har newHar = copy(har);
+        Date savedDateTime = testContext.get(Date.class);
+        if (null != savedDateTime)
+        {
+            newHar = removeTheSameValues(newHar, savedDateTime);
+            testContext.remove(Date.class);
+        }
+        List<HarEntry> harEntries = har.getLog().getEntries();
+        testContext.put(Date.class, harEntries.get(harEntries.size() - 1).getStartedDateTime());
+        return newHar;
+    }
+
+    private Har removeTheSameValues(Har har, Date lastHarEntryDateTime)
+    {
+        har.getLog().getEntries().forEach(entry -> {
+            if (lastHarEntryDateTime.after(entry.getStartedDateTime()) || entry.getStartedDateTime()
+                    .equals(lastHarEntryDateTime))
+            {
+                har.getLog().getEntries().remove(entry);
```

Comment:
is it safe to iterate through collection and remove items from it the same time?

Reply:
the same question for pages